components/resolved_ts/src/scanner.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use engine_traits::KvEngine;
+use kvproto::kvrpcpb::{ExtraOp as TxnExtraOp, IsolationLevel};
+use kvproto::metapb::Region;
+use raftstore::router::RaftStoreRouter;
+use raftstore::store::fsm::{ChangeCmd, ObserveID};
+use raftstore::store::msg::{Callback, SignificantMsg};
+use raftstore::store::RegionSnapshot;
+use tikv::storage::kv::{ScanMode as MvccScanMode, Snapshot};
+use tikv::storage::mvcc::{DeltaScanner, MvccReader, ScannerBuilder};
+use tikv::storage::txn::{TxnEntry, TxnEntryScanner};
+use tokio::runtime::{Builder, Runtime};
+use txn_types::{Key, Lock, TimeStamp};
+
+use crate::errors::{Error, Result};
+
+const DEFAULT_SCAN_BATCH_SIZE: usize = 1024;
+
+pub enum ScanMode {
+    LockOnly,
+    All,
+    AllWithOldValue,
+}
+
+pub struct ScanTask {
+    pub id: ObserveID,
+    pub tag: String,
+    pub mode: ScanMode,
+    pub region: Region,
+    pub checkpoint_ts: TimeStamp,
```

Comment:
```suggestion
```
Seems `checkpoint_ts` is unneeded because we always need to scan all locks

Reply:
ditto